From cab9ca089be320eaab6f176a216abb1c3435d8dd Mon Sep 17 00:00:00 2001
From: "Richard A. Griffiths" <richard.a.griffiths@intel.com>
Date: Wed, 26 Aug 2015 16:34:56 -0700
Subject: [PATCH 11/17] input: silead: fix multitouch id value

The value being passed for multitouch id changes as the points
are moved on the display.  Passing the value of the contact
(finger) rather than the id fixes this.

Change-Id: I83f0bc2f17818a2bdd42034b1415e1444ef415ac
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-20919
Signed-off-by: Richard A. Griffiths <richard.a.griffiths@intel.com>
Reviewed-on: https://android.intel.com:443/472784
---
 drivers/input/touchscreen/silead.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/input/touchscreen/silead.c b/drivers/input/touchscreen/silead.c
index 1f8855c..2517d0e 100644
--- a/drivers/input/touchscreen/silead.c
+++ b/drivers/input/touchscreen/silead.c
@@ -148,8 +148,7 @@ static int silead_ts_request_input_dev(struct silead_ts_data *data)
 static void silead_ts_report_touch(struct silead_ts_data *data, u16 x, u16 y,
 				   u8 id)
 {
-	input_mt_slot(data->input_dev, id - 1);
-	input_report_abs(data->input_dev,ABS_MT_TRACKING_ID, id);
+	input_mt_slot(data->input_dev, id);
 	input_mt_report_slot_state(data->input_dev, MT_TOOL_FINGER, true);
 	input_report_abs(data->input_dev, ABS_MT_POSITION_X, x);
 	input_report_abs(data->input_dev, ABS_MT_POSITION_Y, y);
@@ -212,14 +211,14 @@ static void silead_ts_read_data(struct i2c_client *client)
 					       data->y_max - y : y,
 					       data->x_invert ?
 					       data->x_max - x : x,
-					       id);
+					       i - 1);
 		else
 			silead_ts_report_touch(data,
 					       data->x_invert ?
 					       data->x_max - x : x,
 					       data->y_invert ?
 					       data->y_max - y : y,
-					       id);
+					       i - 1);
 	}
 
 	input_mt_sync_frame(data->input_dev);
-- 
1.9.1

